presto-main/src/main/java/com/facebook/presto/cost/PlanCostEstimate.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cost;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.Double.isNaN;
+
+public final class PlanCostEstimate
```

Comment:
* please add a javadoc and explain difference between `PlanNodeCostEstimate`
* add this class in separate commit, where this class is same as `PlanNodeCostEstimate`

Reply:
Conceptually, there serve the same purpose. They are two separate classes to be able to calculate peak memory in a type safe manner.